<?xml version="1.0"?>
<launch>

	<env name="ROSCONSOLE_FORMAT" value="[${severity}] [${walltime}] [${node}]: ${message}" />

	<arg name="name_tag" default="tag_0" />	<!-- Change also in marker_detector.launch -->
	<arg name="name_file" default="tag_detection_pose.txt" />
	<arg name="topic_tag" default="/camera_network/tag_detections_pose" /> <!-- Same in marker_detector.launch -->
	<arg name="output_topic" default="/arcore/origin" />
	<arg name="centroid_topic" default="/tracker/tracks_smoothed" />
	<arg name="centroid_arcore" default="/arcore/tracks_smoothed" />
	<arg name="skeleton_topic" default="/tracker/skeleton_tracks" />
	<arg name="skeleton_arcore" default="/arcore/skeleton_tracks" />
	<arg name="sensor_name" default="kinect01" />
	<arg name="enable_control" default="false" />

	<!-- Apriltag detector -->
	<include file="$(find optar)/launch/marker_detector.launch">
		<arg name="sensor_name" value="$(arg sensor_name)" />
	</include>


	<!-- Server node for mobile comunication -->
<!--	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch">
		<param name="port" value="9090"/>
	</include>
-->

	<!-- Mobile node publisher -->
	<group if="$(arg enable_control)">
		<node name="mobile_receiver" pkg="optar" type="mobile_receiver" output="screen">
			<param name="input_topic" value="/arcore/vodom"/>
			<param name="output_topic" value="/arcore/marker_array"/>
			<param name="name_tag" value="$(arg name_tag)_arcore"/>
			<param name="origin" value="$(arg output_topic)"/>
			<param name="motion_control" value="enable"/>
		</node>

		<node name="motion_control" pkg="optar" type="motion_control" output="screen">
			<param name="file_path_tag" value="$(find optar)/file/motionControlTag.txt"/>
			<param name="file_path_phone" value="$(find optar)/file/motionControlPhone.txt"/>
			<param name="name_tag" value="tag_1"/>
			<param name="input_topic" value="$(arg topic_tag)"/>
		</node>
	
	</group>

	<group unless="$(arg enable_control)">
		<node name="mobile_receiver" pkg="optar" type="mobile_receiver" output="screen">
			<param name="input_topic" value="/arcore/vodom"/>
			<param name="output_topic" value="/arcore/marker_array"/>
			<param name="name_tag" value="$(arg name_tag)_arcore"/>
			<param name="origin" value="$(arg output_topic)"/>
			<param name="motion_control" value="not_enable"/>
		</node>
	</group>

	<!-- Keep Tag Living  -->
	<node name="keep_tag_living" pkg="optar" type="keep_tag_living" output="screen">
		<param name="file_path" value="$(find optar)/file/$(arg name_file)"/>
		<param name="output_name_tag" value="$(arg name_tag)_arcore"/>
	</node>

	<!-- Tag File -->
	<node name="tag_file" pkg="optar" type="tag_file" output="screen">
		<param name="file_path" value="$(find optar)/file/$(arg name_file)"/>
		<param name="name_tag" value="$(arg name_tag)"/>
		<param name="input_topic" value="$(arg topic_tag)"/>
	</node>

	<!-- Origin sender -->
	<node name="origin_sender" pkg="optar" type="origin_sender" output="screen">
		<param name="name_tag" value="$(arg name_tag)_arcore"/>
		<param name="output_topic" value="$(arg output_topic)"/>
	</node>
	
	<!-- Modifier Centroid -->
	<node name="centroid_modifier" pkg="optar" type="centroid_modifier" output="screen">
		<param name="input_centroid" value="$(arg centroid_topic)"/>
		<param name="origin" value="$(arg output_topic)"/>
		<param name="output_topic" value="$(arg centroid_arcore)"/>
	</node>

	<!-- Modifier Skeleton -->
	<node name="skeleton_modifier" pkg="optar" type="skeleton_modifier" output="screen">
		<param name="input_track" value="$(arg skeleton_topic)"/>
		<param name="origin" value="$(arg output_topic)"/>
		<param name="output_topic" value="$(arg skeleton_arcore)"/>
	</node>
  

</launch>
